Dockerfile

```diff
@@ -40,9 +40,19 @@ RUN mv /tmp/eksctl /usr/local/bin
 COPY gradlew .
 COPY settings.gradle.kts .
 COPY gradlew.bat .
+COPY gradle.properties .
 COPY gradle/ /gradle/
 COPY buildSrc/ /buildSrc/
 COPY validator/ /validator/
 
 # Build gradlew here so that the canary doesn't spend time downloading and building the package
-RUN ./gradlew
\ No newline at end of file
+RUN ./gradlew
```

Comment:
We run `./gradlew` twice in both Dockerfile and workflow. Is it possible to run only once?

Reply:
No unfortunately when we start the container in the workflow the Docker Daemon is in a stopped state, so we have to run it again to initialize it.

Reply:
What if we only run `./gradlew` in workflow without in Dockerfile?

Reply:
The point of running the ./gradlew in Dockerfile is so that we don't have to download the dependencies during our canary run. If we run it in the canary, sometimes it may fail due to transient issues and thereby fail the e2e test. We are trying to migrate as much as possible to the DockerFile to mitigate this issue.

Reply:
By running it in the Dockerfile, we have the build cache saved in our docker image, which we can reuse when we call ./gradlew in the E2E testing.